File: linked_lists/linked_lists.c
# include <stdio.h>
# include <string.h>
# include <stdlib.h>

// arrays's element
struct item
{
    int data;
    struct item *next;
};

struct item *int_array_to_list(int *array, int len)
{
    // return pointer to the first element of the linked list //
    if (!len) {
        return NULL;
    }
    struct item *tmp;
    tmp = malloc(sizeof(*tmp));
    tmp->data = *array;
    tmp->next = int_array_to_list(array + 1, len - 1);
    return tmp;
}

int sum_list(struct item *list)
{
    // return sum of all elements of the list //
    if (list) {
       return list->data + sum_list(list->next); 
    } else {
        return 0;
    }
}

int main(){
    int data_points[] = {1,2,3,4,5,6};
    int len = sizeof(data_points) / sizeof(*data_points);
    struct item *ll = int_array_to_list(data_points, len);
    int sum = sum_list(ll);
    printf("%d\n", sum);
    return 0;
}

File: linked_lists/linked_lists.py
## Столяров, стр. 144 pdf
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    def __repr__(self) -> str:
        return str(self.data)

def int_array_to_list(l)
    ## return pointer to the first element of the linked list ##
    if (len(l) == 0):
        return None
    last = Node(l[-1])
    last.next = int_array_to_list(l[:-1]) 
    return last

def sum_el_ll(first):
    if (first == None):
        return 0
    return first.data + sum_el_ll(first.next)


l = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
first = int_array_to_list(l)
print(sum_el_ll(first))
